XUtils.c

```diff
@@ -78,6 +78,22 @@ void* xReallocArray(void* ptr, size_t nmemb, size_t size) {
    return xRealloc(ptr, nmemb * size);
 }
 
+void* xReallocArrayZero(void* ptr, size_t prevmemb, size_t newmemb, size_t size) {
+   assert((ptr == NULL) == (prevmemb == 0));
+
+   if (prevmemb == newmemb) {
+      return ptr;
+   }
+
+   void* ret = xReallocArray(ptr, newmemb, size);
+
+   if (newmemb > prevmemb) {
+      memset((unsigned char*)ret + prevmemb * size, '\0', (newmemb - prevmemb) * size);
```

Comment:
Missing overflow checks for multiplications `prevmemb * size` and `(newmemb - prevmemb) * size`.

Reply:
`xReallocArray()` in line 88 does an overflow check on `newmemb * size`.
Since `newmemb > prevmemb` is a precondition for this branch `prevmemb * size` is safe.
Since `prevmemb > 0` (it is of type `size_t`) `(newmemb - prevmemb) * size` is also safe.